### ISSUE TITLE
Can't find OpenCL Library on MacOS

### ISSUE THREAD

**Issue by arjunmehta** [unbornchikken/NOOOCL#22]
@unbornchikken Super excited to get this working on my machine. But having an issue:

The package is having trouble finding my OpenCL library. This is similar to/may be the same as #10 but the only way I could get the library to even start tests was to hard code `"/System/Library/Frameworks/OpenCL.framework/OpenCL"` as [the lib path here](https://github.com/unbornchikken/NOOOCL/blob/master/lib/cl11.js#L15).

Maybe this is an issue for [fastcall](https://github.com/cmake-js/fastcall)?

**Comment by unbornchikken** [comment]
I haven't made NOOOCL for Mac, there might be issues. Lemme look into this, I'm sure about we can patch it to work for you. However, the issue is not about fadtcall, I'm sure.

**Comment by unbornchikken** [comment]
Hey, would you try tests of #master?

**Comment by arjunmehta** [comment]
@unbornchikken Unfortunately this still does not work.

Based on two resources, one from [StackOverflow](http://stackoverflow.com/questions/31561309/cmake-warnings-under-os-x-macosx-rpath-is-not-specified-for-the-following-targe) and the other from a very enlightening [CMake Blog post](https://blog.kitware.com/upcoming-in-cmake-2-8-12-osx-rpath-support/) perhaps fastcall needs to add the following to its `CMakeLists.txt` for OSX installs:

```
# enable @rpath in the install name for any shared library being built
# note: it is planned that a future version of CMake will enable this by default
set(CMAKE_MACOSX_RPATH 1)
```

There is a post in the comments on the blog post, but I can't make it out. This stuff confuses me :(

**Comment by arjunmehta** [comment]
Also perhaps of use: [Using Run-Path Dependent Libraries](https://developer.apple.com/library/content/documentation/DeveloperTools/Conceptual/DynamicLibraries/100-Articles/RunpathDependentLibraries.html)

**Comment by unbornchikken** [comment]
Loading libraries has nothing to do with cmake, because those are loaded in runtime instead of compile time. It's dyncall's dynload. I have to check what system calls dynload exactly doing on mac.

**Comment by unbornchikken** [comment]
@arjunmehta according to online articles, your first workaround is actually the solution. :) I don't want to steal your credit, so please make a tiny PR from your patch of that line:

```
this.libName = (process.platform === "win32") ? "OpenCL" : (process.platform === "darwin") ? "/System/Library/Frameworks/OpenCL.framework/OpenCL" : "libOpenCL.so"; 
```

I'll merge and release ASAP.

**Comment by arjunmehta** [comment]
@unbornchikken Added PR https://github.com/unbornchikken/NOOOCL/pull/23

**Comment by unbornchikken** [comment]
Released in v0.12.3
